### PR TITLE
Change fetch to our request utility function.

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Just some quick refactoring before working on #11983

## QA Instructions, Screenshots, Recordings

The reading list section should still be able to archive and unarchive reading list items.

### Archiving

1. Navigate to http://localhost:3000/
2. In the feed, click on the save button of an article in the feed to add it to the reading list.
2. Navigate to http://localhost:3000/readinglist
2. Click on the _Archive_ button of the reading list item.
4. The archived item will disappear from the reading list.
5. A snackbar item at the bottom left of the screen should say "Archiving..." and then disappear after a second or two.

### Unarchiving

1. Navigate to http://localhost:3000/
2. In the feed, click on the save button of an article in the feed to add it to the reading list.
2. Navigate to http://localhost:3000/readinglist
3. Click on the _Archive_ button of the reading list item.
4. The unarchived item will disappear from the archived reading list.
5. A snackbar item at the bottom left of the screen should say "Archiving..." and then disappear after a second or two.
6. Navigate to http://localhost:3000/readinglist/archive
7. Click on the _Unarchive_ button of the archived reading list item.
8. A snackbar item at the bottom left of the screen should say "Unarchiving..." and then disappear after a second or two.

### UI accessibility concerns?

This PR only changes how server data is pulled.

## Added tests?

- [ ] Yes
- [x] No, and this is why: This replaces a native `fetch` function call with our tested stable `request` utility function. We'll be fleshing out end to end tests for the reading list most like related to the work in #11983.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

A woman saying "Let's do this"

### CODE REVIEW

File: app/javascript/readingList/readingList.jsx [hunk 1]
```diff
@@ -88,14 +89,9 @@ export class ReadingList extends Component {
     event.preventDefault();
 
     const { statusView, items, totalCount } = this.state;
-    window.fetch(`/reading_list_items/${item.id}`, {
+    request(`/reading_list_items/${item.id}`, {
```
Comment (nickytonline) [inline app/javascript/readingList/readingList.jsx:92]: This should be awaited and then continue based on a successful request, but that will come in another refactor PR.

File: app/javascript/readingList/readingList.jsx [hunk 2]
```diff
@@ -88,14 +89,9 @@ export class ReadingList extends Component {
...
       method: 'PUT',
-      headers: {
-        'X-CSRF-Token': window.csrfToken,
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({ current_status: statusView }),
-      credentials: 'same-origin',
+      body: { current_status: statusView },
```
Comment (nickytonline) [inline app/javascript/readingList/readingList.jsx:94]: This doesn't need to be stringified as out `request` utility function handles this internally.

https://github.com/forem/forem/blob/master/app/javascript/utilities/http/request.js#L35-L37

**Comment by citizen428** [comment]
@nickytonline Took the liberty of adding this to the "In progress" column of our current cycle board because we agreed to also track the 15 minute fixes.

**Review by citizen428 — APPROVED** [review]

**Review by aitchiss — APPROVED** [review]
LGTM - tested locally and all behaves as expected 👍

**Pull request merged**